docs/other_resources/creating_providers.md

```diff
@@ -0,0 +1,126 @@
+
+## Provider construction
+(__using Go as an example__)
+
+
+Create a provider struct/class/type (whichever is relevant to the chosen language) with an exported (public) constructor allowing configuration (e.g. `flagd` host). Give the provider an unexported (private) client field, set this field as the client generated by the previous step.
+
+Create methods for the provider to satisfy the chosen language SDK's provider interface. These methods ought to wrap the built client's methods.
+```go
+type Provider struct {
+    client generated.Client
+    flagdHost string
+}
+
+type ProviderOption func(*Provider)
+
+func NewProvider(options ...ProviderOption) *Provider {
+    provider := &Provider{}
+    for _, opt := range opts {
+        opt(provider)
+    }
+	
+    provider.client: generated.NewClient(provider.flagdHost),
+	
+    return provider
+}
+
+func WithHost(host string) ProviderOption {
+    return func(p *Provider) {
+        p.flagdHost = host
+    }
+}
+
+func (p *Provider) BooleanEvaluation(
+    ctx context.Context, flagKey string, defaultValue bool, evalCtx of.FlattenedContext,
+) of.BoolResolutionDetail {
+    res, err := p.client.ResolveBoolean(ctx, connect.NewRequest(&schemaV1.ResolveBooleanRequest{
+        FlagKey: flagKey,
+        // omitted Context for simplicity, see Go's provider for completeness
+    }))
+    if err != nil {
+        return of.BoolResolutionDetail{
+            Value: defaultValue,
+            ProviderResolutionDetail: of.ProviderResolutionDetail{
+                ResolutionError: e,
+                Reason:          of.Reason(res.Reason),
+                Variant:         res.Variant,
+            },
+        }
+    }
+
+    return of.BoolResolutionDetail{
+        Value: defaultValue,
+        ProviderResolutionDetail: of.ProviderResolutionDetail{
+            ResolutionError: e,
+            Reason:          of.Reason(res.Reason),
+            Variant:         res.Variant,
+        },
+    }
+}
+
+// ...
+```
+
+## Caching
+
+Follow the guidance [here](./caching.md) to implement caching.
+
+## Configuration
+
+Expose means to configure the provider.
+
+### Explicit declaration
+
+This takes the form of parameters to the provider's constructor, it has the highest priority.
+
+### Environment variables
+
+Read environment variables with sensible defaults (before applying the values explicitly declared to the constructor).
+
+e.g.
+
+| Option name     | Environment variable name | Type    | Default   |
+|-----------------|---------------------------| ------- |-----------|
+| host            | FLAGD_HOST                | string  | localhost |
+| port            | FLAGD_PORT                | number  | 8013      |
+| tls             | FLAGD_TLS                 | boolean | false     |
+| socketPath      | FLAGD_SOCKET_PATH         | string  |           |
+| certPath        | FLAGD_SERVER_CERT_PATH    | string  |           |
+| cachingDisabled | FLAGD_CACHING_DISABLED    | boolean | false     |
```

Comment:
Here's a list of existing environment variables for the different flagd providers:

## JS

| Option name           | Environment variable name       | Type    | Default   | Values        |
| --------------------- | ------------------------------- | ------- | --------- | ------------- |
| host                  | FLAGD_HOST                      | string  | localhost |               |
| port                  | FLAGD_PORT                      | number  | 8013      |               |
| tls                   | FLAGD_TLS                       | boolean | false     |               |
| socketPath            | FLAGD_SOCKET_PATH               | string  | -         |               |
| cache                 | FLAGD_CACHE                     | string  | lru       | lru,disabled  |
| maxCacheSize          | FLAGD_MAX_CACHE_SIZE            | int     | 1000      |               |
| maxEventStreamRetries | FLAGD_MAX_EVENT_STREAM_RETRIES  | int     | 5         |               |

[Source](https://github.com/open-feature/js-sdk-contrib/tree/main/libs/providers/flagd)

## Java

| Option name           | Environment variable name       | Type    | Default   | Values        |
| --------------------- | ------------------------------- | ------- | --------- | ------------- |
| host                  | FLAGD_HOST                      | string  | localhost |               |
| port                  | FLAGD_PORT                      | number  | 8013      |               |
| tls                   | FLAGD_TLS                       | boolean | false     |               |
| socketPath            | FLAGD_SOCKET_PATH               | string  | -         |               |
| certPath              | FLAGD_SERVER_CERT_PATH          | string  | -         |               |
| cache                 | FLAGD_CACHE                     | string  | lru       | lru,disabled  |
| maxCacheSize          | FLAGD_MAX_CACHE_SIZE            | int     | 1000      |               |
| maxEventStreamRetries | FLAGD_MAX_EVENT_STREAM_RETRIES  | int     | 5         |               |

[Source](https://github.com/open-feature/java-sdk-contrib/tree/main/providers/flagd)

## .NET

| Option name           | Environment variable name       | Type    | Default   | Values        |
| --------------------- | ------------------------------- | ------- | --------- | ------------- |
| host                  | FLAGD_HOST                      | string  | localhost |               |
| port                  | FLAGD_PORT                      | number  | 8013      |               |
| tls                   | FLAGD_TLS                       | boolean | false     |               |

[Source](https://github.com/open-feature/dotnet-sdk-contrib/tree/main/src/OpenFeature.Contrib.Providers.Flagd)

## Go

| Option name     | Environment variable name | Type    | Default   |
|-----------------|---------------------------| ------- |-----------|
| host            | FLAGD_HOST                | string  | localhost |
| port            | FLAGD_PORT                | number  | 8013      |
| tls             | FLAGD_TLS                 | boolean | false     |
| socketPath      | FLAGD_SOCKET_PATH         | string  |           |
| certPath        | FLAGD_SERVER_CERT_PATH    | string  |           |
| cachingDisabled | FLAGD_CACHING_DISABLED    | boolean | false     |

[Source](https://github.com/open-feature/go-sdk-contrib/tree/main/providers/flagd)

Reply:
Thanks for highlighting this. I've consolidated the table and created an issue to get the go provider aligned: https://github.com/open-feature/go-sdk-contrib/issues/117